### ISSUE TITLE
No progress/feedback when restoring backup from onboarding

### ISSUE THREAD

**Issue by bschatzow** [home-assistant/frontend#11854]
### Describe the issue you are experiencing

Had to do a system recovery.  Did a reformat and did an onboarding restore. It picks the correct image and then it loads the first screen again.  I have no idea if it is doing anything.

### What operating system image do you use?

rpi4-64  (Raspberry Pi 4/400 64-bit OS)

### What version of Home Assistant Operating System is installed?

7.4

### Did you upgrade the Operating System.

No

### Steps to reproduce the issue

1. Load the latest RPI with etcher
2. Go to Onboarding and pick image
3.
...

### Anything in the Supervisor logs that might be useful for us?

```txt
N/A
```

### Anything in the Host logs that might be useful for us?

```txt
N/A
```

### System Health information

N/A

### Additional information

![image](https://user-images.githubusercontent.com/36886571/155606586-0d338eeb-3176-4f87-8cf2-1954c63aed86.png)

**Comment by bschatzow** [comment]
I did open this last year an someone closed it saying that this was not an issue. Thanks for confirming. Fixing these little things makes HA better for all

**Comment by bschatzow** [comment]
Just tried it again due to an issue with 8.0 RC2.  I think the restore got worst.  There is not an onboarding option and now after the backup screen comes up and I upload the backup and hit restore I go to this:
![image](https://user-images.githubusercontent.com/36886571/163651742-c4754810-aba7-4c11-9ef0-fc5101a5c828.png)
and connection lost.  No idea on what is going on.

**Comment by tgambee** [comment]
I am experiencing the same behavior.  No luck with the restore.  Steps exactly as described by @bschatzow.  maybe it's running, but no indication that it is and nothing ever changes.

**Comment by ericleigh007** [comment]
Yes, definitely a problem here as well.
User tries to restore, has no idea what is going on, and then finally gives up and powers down the device.
Was it restoring when that happened?   Is he likely to get some partially-restored thing?
Should it actually automatically reload the page?

In my case, no restore window opened after I said "sure I want to restore the full backup."  In this case, the login screen was just re-displayed.

I seem to have verified that something was going on that had things tied up.  I tried to restore the backup again, and it reported "error fetching backup".

I do not believe we can consider this works at all.  What's the trick to understand what's happening?

At least if we publish something indicating how to use the restore successfully, maybe that'll be at least some progress.

**Comment by antoweb** [comment]
For me this is a very problem because only in supervisor log you can see the restor progress

**Comment by sEdsKi** [comment]
Just experienced this. As described by previous posts. Waiting on a "this site can't be reached" screen hoping the restore is progressing.

Running the restore on a NUC install.

**Comment by stevem405** [comment]
I'm also seeing the same problem
- Full backup (32 bit)
- haos_rpi3-64-9.4.img.xz install
- Full backup > Restore (without creating a new user & PW
No indication anything is happening

**Comment by viseshrp** [comment]
Ran into the same issue few minutes ago trying to migrate from a 2gb Pi 4b to an 8gb. Took a backup on the 2gb, tried to restore on the 8gb during onboarding after installing from here:
https://github.com/home-assistant/operating-system/releases/download/9.4/haos_rpi4-64-9.4.img.xz
No progress, nothing. 
What's worse is that refreshing still takes me to the onboarding screen. 
Onboarded manually and tried restoring manually from the Backups page, lost connection for a bit, refreshed and still nothing.

**Comment by LucaZiegler** [comment]
Same issue after I uploaded a backup on a fresh home assistant os:
![image](https://github.com/home-assistant/frontend/assets/36471502/784de225-9a3f-4cbe-bbf1-e2562026a730)
![image](https://github.com/home-assistant/frontend/assets/36471502/74831f7c-41a9-4141-95c7-69f65d7b004c)

**Comment by requiemmg** [comment]
This is really annoying, please do something about this asap...Tried to restore a backup from onboarding, was thrown back to login/user generation prompt without further notice and noticed that nothing has happened hours later. Please provide ANY kind of restore progress, at least on the observer or sth like that...

**Comment by GWL20** [comment]
Yes, no improvement with the latest 2023.7.3 version. Is there a workaround for determining when the restore has finished?

**Comment by airampg** [comment]
Experiencing the same issue, and hoping it can be resolved soon. Thank you to the devs for their hard work.

**Comment by ChrisCarini** [comment]
I believe https://github.com/home-assistant/supervisor/pull/4503 will help work towards resolving this issue, correct @mdegat01 ?

**Comment by drled** [comment]
Experiencing the same issue

**Comment by pedroportasvieira** [comment]
Doing a restore from onboarding now, due to a system failure, and same issue... it would be really nice to have feedback.

**Comment by lovekull76** [comment]
> Doing a restore from onboarding now, due to a system failure, and same issue... it would be really nice to have feedback.

What version? 2023.9 did introduce a new Onboarding GUI.
I'm doing a HW swap soon and it would be nice if this issue was fixed in .9

**Comment by ve6rah** [comment]
Now it seems to go to a spinning wheel that says "Restore in progress", However after an hour of that, I still have no indication if it's actually doing anything.

I was told that supervisor logs should show something, but when I check them from the console, I see nothing in the past hour, so either something is wrong, or they don't show anything during the restore.

I saw someone else say to use "top" to see what processes are running, but they failed to say what process to look for,  I see 168 tasks running, and none of them say "restoring backup", so I have no idea.

**Comment by lovekull76** [comment]
For me the restore of the TimeScaleDB add on took about 8h due to a very big database. I you have access to the backup file, expand it and look in the .json file and check the order of the add ones in the list.
If you check the supervisor log and find the last add on it restored, I think you can figure out which add on it is working on at the moment from the list. Then check the size of that add on in the expanded backup, if it is something like 10GB+ it will take a looooong time to restore.
This gave me peace of mind and I waited it out. It could have been a lucky coincidence that the list in the json was actually the order of add on restore.

**Comment by ve6rah** [comment]
> Check the supervisor log and find the last add on it restored. ThenI think you can figure out which add on it is working on at the moment from the list.

In my case, the supervisor log never showed anything at all. No indication that the system was doing anything, and no errors. After several hours of waiting, I forced a reboot and started the setup from scratch. Turns out, that my backup had a password, and I had forgotten to enter it. I find it somewhat inexcusable that no error was shown for that situation.

That said, even after that, I did a system restore properly, my 4 GB backup took over 9 hours to restore, and when it was done the database which is the reason it was so large is corrupted and didn't work anyway.

What I really found odd though, is even restoring a partial backup, and choosing a single add-on, takes over an hour to restore. I just can't come up with any possible excuse for that amount of time being used.

All told, restoring my system, took me about 20 hours. Had I simply copy and pasted the config files, and installed the rest from scratch, I'm pretty sure I would have been done in a quarter of the time.

There is something extremely broken about the backup and restore functionality of home assistant!

**Comment by lovekull76** [comment]
> That said, even after that, I did a system restore properly, my 4 GB backup took over 9 hours to restore, and when it was done the database which is the reason it was so large is corrupted and didn't work anyway.

Yes, found that out the hard way too. I had to stop the db add on and make a partial backup then a partial restore. My db is about 10GB. Doing a db dump from within pgadmin took about 3 minutes … 

I have now setup an automated partial backup of HA minus db add on and a pgagent job to do a db dump Ian’s complement.

**Comment by DanielHabenicht** [comment]
Just adding that the restore with password experience is even worse, if you supply the wrong password: 
It simply does not show the error and gives you the impression that it is restoring your backup, only after you look into the supervisor logs an hour later it shows you "Invalid password for backup <id>". 
That's really annoying.

**Comment by shadowpho** [comment]
+1, this is terrible. Transferring small backup (200MB) into a fast device (rasp pi4 8gb, fast sd card). No changer even after 2 hours. Would be nice if we could enable ssh and at least check logs to see if it’s stuck or just slow.

**Comment by agners** [comment]
This essentially also got reported in #19388 and really should be fixed with #19389.
